src/main/java/seedu/commands/FilterCommand.java

```diff
@@ -20,49 +34,104 @@ public InternshipList getFilteredInternships() {
+
         if (args.isEmpty()) {
             uiCommand.showInsufficientArguments();
             return;
         }
 
-        if (args.size() > 1) {
-            uiCommand.showOutput("Too many flags provided. Can only filter by one flag at a time");
+        for (String arg : args) {
+            String[] words = arg.split(" ", 2);
+            try {
+                executeFilterByOneFlag(words);
+            } catch (IllegalArgumentException e) {
+                return;
+            }
+        }
+
+        functionComplete = true;
+        filteredInternships.listAllInternships();
+    }
+
+    private void executeFilterByOneFlag(String[] words) {
+        String flag = words[INDEX_FIELD];
+        // Retrieve the corresponding getter method based on the flag
+        InternshipFieldGetter getter = fieldGetters.get(flag);
+
+        if (getter == null) {
+            uiCommand.clearInvalidFlags();
+            uiCommand.addInvalidFlag(flag);
+            uiCommand.printInvalidFlags();
+            throw new IllegalArgumentException();
+        }
+
+        if (words.length < 2) {
+            uiCommand.showOutput(words[INDEX_FIELD] + " field cannot be empty");
+            throw new IllegalArgumentException();
+        }
+
+        String searchTerm = words[INDEX_DATA];
+        BiPredicate<YearMonth, YearMonth> dateComparator = null;
+
+        switch (flag) {
+        case "role":
+        case "company":
+            filterByRoleAndCompany(getter, searchTerm);
             return;
+
+        case "from":
+            dateComparator = YearMonth::isBefore;
+            break;
+
+        case "to":
+            dateComparator = YearMonth::isAfter;
+            break;
+
+        default:
+            assert false: "Should never be able to reach this statement if all flags are accounted for";
         }
 
-        String[] words = args.get(0).split(" ", 2);
-        String flag = words[0];
-        try {
-            // Map flags to getter methods using lambdas
-            Map<String, InternshipFieldGetter> fieldGetters = new HashMap<>();
-            fieldGetters.put("role", Internship::getRole);
-            fieldGetters.put("company", Internship::getCompany);
-            fieldGetters.put("from", Internship::getStartDate);
-            fieldGetters.put("to", Internship::getEndDate);
-
-            // Retrieve the corresponding getter method based on the flag
-            InternshipFieldGetter getter = fieldGetters.get(flag);
-
-            if (getter == null) {
-                uiCommand.clearInvalidFlags();
-                uiCommand.addInvalidFlag(flag);
-                uiCommand.printInvalidFlags();
-                return;
-            }
+        if (isValidDate(flag, searchTerm)) {
+            filterByDate(getter, searchTerm, dateComparator);
+        }
+    }
 
-            String searchTerm = words[1];
+    private boolean isValidDate(String flag, String searchTerm) {
+        if (!searchTerm.matches("\\d{2}/\\d{2}")) {
```

Comment:
I think it might be better to attempt to parse the date and see if it returns an error. This will accept something like 40/24 which is not a valid date.